### PR TITLE
[Metrics API] Remove unnecessary auto trait implementations

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Remove unnecessary `Ord` and `PartialOrd` implementation for `KeyValue`
- This was only used for sorting a vec of `KeyValue` pairs in the SDK which could be done using `sort_unstable_by` method anyway

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/mod.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ fn calculate_hash(values: &[KeyValue]) -> u64 {
 
 impl AttributeSet {
     fn new(mut values: Vec<KeyValue>) -> Self {
-        values.sort_unstable();
+        values.sort_unstable_by(|a, b| a.key.cmp(&b.key));
```
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/mod.rs:107]: will this be a breaking change if the user is sorting the KeyValue using the sort/sort_unsable before calling the metrics API?

Comment (utpilla) [reply opentelemetry-sdk/src/metrics/mod.rs:107]: Yes, but I don't expect anyone to be doing that. We don't ask to sort the metric attributes before recording a measurement.

Comment (lalitb) [reply opentelemetry-sdk/src/metrics/mod.rs:107]: Yes agree. Good to have changlog entry just in case someone is doing that :)

**Review by lalitb — APPROVED** [review]
LGTM with changelog

File: opentelemetry-sdk/src/metrics/mod.rs (hunk 1 above)
Comment (utpilla) [reply opentelemetry-sdk/src/metrics/mod.rs:107]: Updated.

**Pull request merged**